src/execution_plan/ops/op_index_scan.c

```diff
@@ -0,0 +1,58 @@
+#include "op_index_scan.h"
+
+OpBase *NewIndexScanOp(Graph *g, Node **node, IndexIterator *iter) {
+  return (OpBase*)NewIndexScan(g, node, iter);
+}
+
+IndexScan* NewIndexScan(Graph *g, Node **node, IndexIterator *iter) {
+
+  IndexScan *indexScan = malloc(sizeof(IndexScan));
+  indexScan->node = node;
+  indexScan->_node = *node;
+  indexScan->iter = iter;
+
+  // Set our Op operations
+  indexScan->op.name = "Index Scan";
+  indexScan->op.type = OpType_INDEX_SCAN;
+  indexScan->op.consume = IndexScanConsume;
+  indexScan->op.reset = IndexScanReset;
+  indexScan->op.free = IndexScanFree;
+  indexScan->op.modifies = NewVector(char*, 1);
+
+  Vector_Push(indexScan->op.modifies, Graph_GetNodeAlias(g, *node));
+
+  return indexScan;
+}
+
+OpResult IndexScanConsume(OpBase *opBase, Graph* graph) {
+  IndexScan *op = (IndexScan*)opBase;
+
+  if(op->iter == NULL) {
```

Comment:
How's that posible?
Maybe assert within NewIndexScanOp

Reply:
You're right, this case should never be possible (especially now that we're building string and numeric skiplists for each indexed property).

I've deleted this check. Do you want an explicit assert, or does this assumption seem safe enough?